### PR TITLE
Auto provision

### PR DESCRIPTION
**Pull request by tillrohrmann**
Adds a feature to disable the auto-provisioning of partitions by the cluster controller. To disable the auto-provisioning, you can set the `auto-provision-partitions = false` in the configuration or on the CLI via `--auto-provision-partitions=false`.

This PR is based on #2053

### CODE REVIEW

File: crates/admin/src/cluster_controller/service.rs
Comment (AhmedSoliman) [inline crates/admin/src/cluster_controller/service.rs:269]: Not a blocker, but I'd have loved if we rely more on Live configuration for a more reactive experience.

File: crates/admin/src/cluster_controller/service.rs
Comment (AhmedSoliman) [inline crates/admin/src/cluster_controller/service.rs:340]: nit: Might be worth logging that we are initializing

File: crates/node/src/lib.rs
Comment (AhmedSoliman) [inline crates/node/src/lib.rs:303]: what should we expect to do here?

File: crates/types/src/config/cli_option_overrides.rs
Comment (AhmedSoliman) [inline crates/types/src/config/cli_option_overrides.rs:90]: I think we will need to review the ergonomics of this process but I wouldn't block merging this.

**Review by AhmedSoliman — APPROVED** [review]
Very nice. I'm fine with this as a temporary design until we discuss a more ergonomic way for single-node vs. cluster setup.

File: crates/types/src/config/cli_option_overrides.rs
Comment (tillrohrmann) [reply crates/types/src/config/cli_option_overrides.rs:90]: Sounds good :-)

File: crates/node/src/lib.rs
Comment (tillrohrmann) [reply crates/node/src/lib.rs:303]: This is not fully clear to me yet. It could be the initial default log provider or whether to auto provision partitions if this needs to be agreed upon.

File: crates/admin/src/cluster_controller/service.rs
Comment (tillrohrmann) [reply crates/admin/src/cluster_controller/service.rs:340]: Yes, will add it.

File: crates/admin/src/cluster_controller/service.rs
Comment (tillrohrmann) [reply crates/admin/src/cluster_controller/service.rs:269]: For what it's worth, the service use the `Live<Configuration>` and here we are only using a snapshot to setup the `LogsController`. Apart from this place, the `LogsController` does not need access to the `Configuration` so far.

**Comment by tillrohrmann** [comment]
Thanks for the review @AhmedSoliman. I will address your comments and then merge this PR once it gives green light.

**Pull request merged**
